.eleventy.js

```diff
@@ -140,33 +154,59 @@ module.exports = function (config) {
   let markdownLibrary = markdownIt({
     html: true,
     breaks: true,
-    linkify: true
-  }).use(markdownItAnchor, {
-    permalink: markdownItAnchor.permalink.ariaHidden({
-      placement: 'after',
-      class: 'direct-link',
-      symbol: '',
-      level: [1, 2, 3, 4],
-    }),
-    slugify: config.getFilter('slug'),
-  }).use(markdownItTable);
+    linkify: true,
+    typographer: true
+  })
+    .use(function (md) {
+      // Override the default link rendering rule to add target and rel attributes
+      const defaultRender = md.renderer.rules.link_open || function (tokens, idx, options, env, self) {
+        return self.renderToken(tokens, idx, options);
+      };
+      md.renderer.rules.link_open = function (tokens, idx, options, env, self) {
+        const token = tokens[idx];
+        const hrefAttr = token.attrs ? token.attrs.find(attr => attr[0] === 'href') : null;
+        if (hrefAttr && /^https?:\/\//.test(hrefAttr[1])) {
+          token.attrSet('target', '_blank');
+          token.attrSet('rel', 'noopener noreferrer');
+        }
+        return defaultRender(tokens, idx, options, env, self);
+      };
+      md.renderer.rules.code_inline = (tokens, idx, {langPrefix = ''}) => {
+        const token = tokens[idx];
+        return `<code class="${langPrefix}plaintext">${htmlEntities(token.content)}</code>&nbsp;`;
+      };
+    })
+    .use(markdownItAnchor, {
+      permalink: markdownItAnchor.permalink.ariaHidden({
+        placement: 'after',
+        class: 'direct-link',
+        symbol: '',
+        level: [1, 2, 3, 4],
+      }),
+      slugify: config.getFilter('slug'),
+    })
+    .use(markdownItTable);
+
   config.addFilter("markdown", (content) => {
     return markdownLibrary.render(content);
   });
-  markdownLibrary.renderer.rules.link_open = function (tokens, idx, options, env, self) {
-    const href = tokens[idx].attrGet("href");
-    if (href && href.startsWith("http")) {
-      tokens[idx].attrPush(["target", "_blank"]);
-      tokens[idx].attrPush(["rel", "noopener noreferrer"]);
-    }
-    return self.renderToken(tokens, idx, options);
-  };
-  markdownLibrary.renderer.rules.code_inline = (tokens, idx, {langPrefix = ''}) => {
-    const token = tokens[idx];
-    return `<code class="${langPrefix}plaintext">${htmlEntities(token.content)}</code>&nbsp;`;
-  };
   config.setLibrary('md', markdownLibrary);
 
+  config.addTransform("externalLinks", (content, outputPath) => {
+    if (outputPath && outputPath.endsWith(".html")) {
+      const dom = new JSDOM(content);
+      const links = dom.window.document.querySelectorAll("a[href^='http']");
+
+      links.forEach(link => {
```

Comment:
oh maybe this is where to put the external link class? not the above one?

Reply:
Should be both.  One is for markdown and the other is for html files.

Reply:
Okay, I'll revert this change.